### ISSUE TITLE
Verify that the database connection module actually uses the system trust store

### ISSUE THREAD

**Issue by wesley-dean-gsa** [GSA/site-scanning#135]
Per [a thread in Slack](https://gsa-tts.slack.com/archives/C0A1Z7L2U/p1636572901011300), it would be helpful if we verified how the libraries we're using (e.g., NestJS w/ TypeORM) are actually using the system trust store.  The error what we were seeing about there being a self-signed certificate should not be something we're seeing provided the underlying libraries are properly using the system trust store.  Therefore, the workaround we put in place for [129](https://github.com/GSA/site-scanning-engine/pull/129) should not be necessary.

So, we should document exactly what libraries are being used and verify that they use the system trust store correctly.

**Comment by wesley-dean-gsa** [comment]
When importing the CodeQL stuff (GSA/site-scanning-engine#158) the "disable certificate validation" we added for GSA/site-scanning-engine#129 was flagged as a bad thing that we should correct.  Ben Berry pointed out that if the library we're using were to use the system's trust store, this wouldn't be an issue.

**Comment by wesley-dean-gsa** [comment]
This is a little bit more challenging than I expected.

First, I made a comment referencing GSA/site-scanning-engine#129 as being related to this.  It is not.  The line that was flagged had to deal with a scan to see whether or not a randomly-generated path on a given hostname pointed to a page or not -- it didn't related to Postgres or Redis interactions as I had first suspected.  GSA/site-scanning-engine#146 addresses this concern by annotating the line in question to be ignored by CodeQL.

Second, researching how the libraries we use to interact with external tooling via SSL / TLS brings up many examples of where libraries request the path to local certs and CAs.  Some platforms use a single file with all of the certs, some use a directory with many files while some platforms store files in different locations, etc..  All that's fine.. problem is, specifying the location of the file(s) to use circumvents the system from using its own logic and configuration..  which is a challenge.

So, long story short, I'm going to continue to look for how to work around this and document what I find.

**Comment by wesley-dean-gsa** [comment]
We discussed this during sprint planing on Tuesday, February 8th.  The status is that we're in a stable place (e.g., nothing is broken), we're in compliance with the ZTA (Zero-Trust Architecture) memo in that we use encryption between endpoints, and the systems with which we're interacting are internal and brokered by the cloud.gov platform.

I judge that this issue, while unresolved at the moment, is a lower priority than some other issues.  Moreover, the resolution may involve modifying upstream libraries and/or making implementation-specific changes (e.g., codify or parameterize the location of the relevant certificates, CAs, etc.).

Therefore, I support moving this to the backlog as technical debt to be addressed at a later date.

**Comment by danielnaab** [comment]
Recommend keeping this on the backlog as a low-priority item. Summary of this issue: TLS/SSL is configured in the application and used by the `node-postgres` library (via TypeORM). However, certificate issues are ignored, because node-postgres is not using the system trust store. It may be possible to point the configuration at the certificate in /etc/ssl/certs, where the appropriate RDS certificate should reside on the app container: https://node-postgres.com/features/ssl #247

**Comment by gbinal** [comment]
(moving to [the backlog](https://github.com/GSA/site-scanning/issues/337) for now)
